### PR TITLE
Py-opentuner: new package

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-opentuner/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    homepage = "http://opentuner.org/"
+    url      = "https://github.com/jansel/opentuner"
+
+    version('0.8.0', git='https://github.com/jansel/opentuner', commit='4cb9135')
+
+    # No support for Python 3 yet
+    depends_on('python@2.7:2.8')
+
+    depends_on('py-sqlalchemy@0.8.2:')
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-opentuner/package.py]: you could order these by name. This makes finding a specific dependency simpler (think adding a new version with different dependencies)

File: var/spack/repos/builtin/packages/py-opentuner/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    depends_on('py-numpy@1.8.0:')
+    depends_on('py-argparse@1.2.1:')
+    depends_on('py-pysqlite@2.6.3:')
+    depends_on('py-fn@0.2.12:')
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-opentuner/package.py]: There are optional dependencies (`optional-requirements.txt`)
```
django==1.6.1
matplotlib==1.1.1
virtualenv==1.9.1
```
not sure if it's worth adding these as variants though.

File: var/spack/repos/builtin/packages/py-opentuner/package.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    """An extensible framework for program autotuning."""
+
+    homepage = "http://opentuner.org/"
+    url      = "https://github.com/jansel/opentuner"
+
+    version('0.8.0', git='https://github.com/jansel/opentuner', commit='4cb9135')
+
+    # No support for Python 3 yet
+    depends_on('python@2.7:2.8')
+
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-opentuner/package.py]: can you add
```
depends_on('py-setuptools', type='build')
depends_on('py-pandoc', type='build')
```
While not strictly necessary, `setup.py` tries to import both

File: var/spack/repos/builtin/packages/py-opentuner/package.py (hunk 2 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-opentuner/package.py]: I think there is no django package in spack, and adding matplotlib alone doesn't make sense as the virtualization tool requires both. virtualenv isn't really necessary in spack, right?

File: var/spack/repos/builtin/packages/py-opentuner/package.py (hunk 3 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-opentuner/package.py]: There is no (py-)pandoc package, but I added py-setuptools.

File: var/spack/repos/builtin/packages/py-opentuner/package.py (hunk 1 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-opentuner/package.py]: Done, thanks.

File: var/spack/repos/builtin/packages/py-opentuner/package.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    depends_on('py-argparse@1.2.1:')
+    depends_on('py-fn@0.2.12:')
+    depends_on('py-numpy@1.8.0:')
+    depends_on('py-pysqlite@2.6.3:')
+    depends_on('py-setuptools', type='build')
+    depends_on('py-sqlalchemy@0.8.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-opentuner/package.py]: All of these dependencies other than `py-setuptools` should be `type=('build', 'run')`.

Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-opentuner/package.py]: Done, thanks.

**Pull request merged**